xonsh/completer.py

```diff
@@ -157,20 +199,19 @@ def complete(self, prefix, line, begidx, endidx, ctx=None):
                     s = s.rstrip() + slash
                 rtn.add(s)
             if len(rtn) == 0:
-                rtn = self.path_complete(prefix)
-            return sorted(rtn)
+                rtn = self.path_complete(prefix, path_str_start, path_str_end)
         elif prefix.startswith('${') or prefix.startswith('@('):
             # python mode explicitly
             rtn = set()
         elif prefix.startswith('-'):
-            return sorted(self._man_completer.option_complete(prefix, cmd))
+            return lprefix, sorted(self._man_completer.option_complete(prefix, cmd))
```

Comment:
Yep.  I had dreams that `lprefix` would be used in both, but unless I'm missing something, readline's completer is much more finicky.

It represents the length of the prefix string (how far back do we have to delete to get to the front of the completion).